GPT/gpt-function-calling.py

```diff
@@ -0,0 +1,229 @@
+from talon import Module, actions, app, settings
+from typing import Callable, Literal
+import webbrowser
+import tempfile
+import requests
+import os
+import json
+
+mod = Module()
+
+mod.setting(
+    "openai_model", type=Literal["gpt-3.5-turbo", "gpt-4"], default="gpt-3.5-turbo"
+)
+
+# Defaults to Andreas's custom notifications if you have them installed
+def notify(message: str):
+    try:
+        actions.user.notify(message)
+    except:
+        app.notify(message)
+    # Log in case notifications are disabled
+    print(message)
+
+
+def gpt_query(prompt: str, content: str, insert_response: Callable[[str], str]) -> str:
```

Comment:
even though this is in a separate namespace let's rename it, to `gpt_function_query` or something similar just so we don't get confused by clashing the function names.